### PR TITLE
Idempotent build: configure.php

### PR DESCRIPTION
**Pull request by alfsb**
This PR does a few clean ups, creates functions for consistent loading and reloading of manual XML saves all intermediary and final XML on `temp/`, and changes error reporting.

The reloading of XML is necessary to validation and file/line/column information works correctly, The error message causes build failures like this:
```
 Opening and ending tag mismatch: para line 19 and par
```
to be reported like this:
```
[FATAL  26:10] Opening and ending tag mismatch: para line 19 and par
[FATAL  27:1] chunk is not well balanced
[FATAL en/appendices/migration56/incompatible.xml 87:42] Entity 'migration56.openssl.peer-verification' failed to parse
[FATAL en/appendices/migration56/incompatible.xml 144:1] chunk is not well balanced
[FATAL en/appendices/migration56.xml 24:39] Entity 'appendices.migration56.incompatible' failed to parse
[FATAL en/appendices/migration56.xml 54:1] chunk is not well balanced
[FATAL zpr-doc-base/manual.xml 443:27] Entity 'appendices.migration56' failed to parse
```

That's right. We now have contextual, call stack-like error messages for errors *inside* of entities. How nice is this?

### CODE REVIEW

File: configure.php [hunk 1]
```diff
@@ -201,40 +202,44 @@ function make_scripts_executable($filename) // {{{
...
+
+    foreach( $errors as $error )
+    {
+        $mssg = rtrim( $error->message );
+        $file = $error->file;
+        $line = $error->line;
+        $clmn = $error->column;
+
+        if ( str_starts_with( $mssg , 'XPointer evaluation failed:' ) && ! $report )
+            continue; //
```
Comment (Girgias) [inline configure.php:229]: Missing comment?

File: configure.php [hunk 2]
```diff
@@ -914,13 +922,12 @@ function getFileModificationHistory(): array {
...
-$dom->save($mxml);
-$dom->load($mxml, $LIBXML_OPTS);
+dom_saveload( $dom );   // idempotent path
+$dom->save($mxml);      // non idempotent, historical path
 if ($dom->relaxNGValidate(RNG_SCHEMA_FILE)) {
     echo "done.\n";
     printf("\nAll good. Saving %s... ", basename($ac["OUTPUT_FILENAME"]));
-    flush();
-    $dom->save($mxml);
+    $dom->save($mxml); // save it again... does validations changes somethnig?
```
Comment (Girgias) [inline configure.php:930]: No, this is useless actually, and could probably be removed in a different PR/commit immediately.

**Review by Girgias — COMMENTED** [review]
This is pretty cool!

File: configure.php (hunk 1 above)
Comment (alfsb) [reply configure.php:229]: Yeah. Fixed.

File: configure.php (hunk 2 above)
Comment (alfsb) [reply configure.php:930]: I will do some some tests to be certain. before removing it in this PR.

Comment (alfsb) [reply configure.php:930]: The files are identical, as expected. Removing the redundant save(), message changed a little.

**Comment by alfsb** [comment]
Merged. The CI errors are unrelated as far I can see.

`git mv` reference/spl/countable.xml to language/predefined/countable.xml and update the entity reference at end of XML.
